node/bin/src/revm_consistency_checker/revm_state_db.rs

```
@@ -0,0 +1,146 @@
+        Ok(self
+            .state
+            .state_view_at(self.latest_block)?
+            .get_account(address)
+            .map(|props| {
+                let code_hash = B256::from(props.bytecode_hash.as_u8_array());
+
+                AccountInfo {
+                    nonce: props.nonce,
+                    balance: props.balance,
+                    code_hash: B256::from(props.observable_bytecode_hash.as_u8_array()),
+                    code: if props.bytecode_hash.is_zero() {
+                        None
+                    } else {
+                        let bytecode = self.code_by_hash_ref(code_hash).expect("code_by_hash");
+                        Some(get_unpadded_code(bytecode.bytes_slice(), &props).into())
+                    },
+                }
+            }))
+    }
+
+    /// Gets account code by its hash.
+    fn code_by_hash_ref(&self, code_hash: B256) -> Result<Bytecode, Self::Error> {
+        Ok(self
+            .state
+            .state_view_at(self.latest_block)?
+            .get_preimage(code_hash)
+            .map(|bytes| Bytecode::new_raw(bytes.into()))
+            .unwrap_or_default())
+    }
+
+    /// Gets storage value of address at index.
+    fn storage_ref(
+        &self,
+        address: Address,
+        index: StorageKey,
+    ) -> Result<StorageValue, Self::Error> {
+        let flat_key = derive_flat_storage_key(
+            &B160::from_be_bytes(address.into_array()),
+            &fixed_bytes_to_bytes32(index.into()),
+        );
+        Ok(self
+            .state
+            .state_view_at(self.latest_block)?
+            .read(flat_key)
+            .unwrap_or_default()
+            .into_u256_be())
+    }
+
+    /// Gets block hash by block number.
+    fn block_hash_ref(&self, number: u64) -> Result<B256, Self::Error> {
+        if let Some(diff) = self.latest_block.checked_sub(number)
+            && diff < 256
```

shouldn't we return an error if diff > 256 ?